### ISSUE TITLE
Check does not check the entire repository 

### ISSUE THREAD

**Issue by SiddharthManthan** [garethgeorge/backrest#397]
This is continuation of #75.

## Problem
- Currently check randomly selects a part of repository.
- This does not guarantee the all the data will be check after sufficient runs. ([Source](https://restic.readthedocs.io/en/latest/045_working_with_repos.html#:~:text=This%20will%20not%20guarantee%20to%20cover%20all%20available%20pack%20files%20after%20sufficient%20runs%2C%20but%20it%20is%20easy%20to%20automate%20checking%20a%20small%20subset%20of%20data%20after%20each%20backup.))

## Solution
- Implement `--read-data-subset=n/t` ([Documentation](https://restic.readthedocs.io/en/latest/045_working_with_repos.html#:~:text=%2D%2Dread%2Ddata%2Dsubset%3Dn/t)).
- It will check all the data after sufficient runs.

I have filed it as bug report because the UI currently does not mention that all data is not checked. There should be a warning or proper check support should be implemented.

**Comment by Thinkscape** [comment]
Does this repo setting not fit your use case?
Setting this to 100% will re-download the whole thing ...

<img width="479" alt="image" src="https://github.com/user-attachments/assets/04b999d7-c3cb-4c68-8d60-892f4a93249a">

**Comment by qubist-pixel-ux** [comment]
> Does this repo setting not fit your use case? Setting this to 100% will re-download the whole thing ...
> 
> <img alt="image" width="479" src="https://private-user-images.githubusercontent.com/270528/351139892-04b999d7-c3cb-4c68-8d60-892f4a93249a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MjcxMzMsIm5iZiI6MTcyMTcyNjgzMywicGF0aCI6Ii8yNzA1MjgvMzUxMTM5ODkyLTA0Yjk5OWQ3LWMzY2ItNGM2OC04ZDYwLTg5MmY0YTkzMjQ5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyM1QwOTI3MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hOGVlNmZlMmY0ZjM0ODE5YmUwM2ZkMDg5ZWNkZTZjMGI3YjY1YjIxYTg2ZjNkYWU2OWUwOWQxODFmN2I5MzRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._4vrvwOD2no9w1wZmrQhakDgBMEJQd8953HzELRwTJM">

The repo is quite large. Downloading all the data will take a long time.

**Comment by garethgeorge** [comment]
hey, `--read-data-subset=n/t` is only guaranteed to check the whole repo if you systematically increment `n` which isn't something I expect to implement as that value would need to be tracked somewhere which adds complexity relative to the value of the feature.

The alternative would be checking random chunks of the repo, but this likely isn't better than using a random percentage.

**Comment by Thinkscape** [comment]
> Check not only checks the integrity of repository files, but also checks for repository corruption.

That is true even in the current version of backrest. If you had set it to `0.001%` (`0%` might work as well) it already checks the indexes, and stats the packs.


<img width="512" alt="image" src="https://github.com/user-attachments/assets/d4da03fe-20b1-4c88-b87b-0dd2c67179a8">
<img width="497" alt="image" src="https://github.com/user-attachments/assets/98703b9c-77d1-4170-b9ce-4f9d1e7363ab">

**Comment by sedlund** [comment]
It seems that using a percentage will eventually check every pack with probability. For example, at 25%, it will most likely check every pack after _at most_ six runs.

I would suggest that if you are interested in this feature that it would be be easier to figure that out than to implement this request.

Additionally, if using subset=n/t  - the complete operation (checking all sets) must be completed within a short time frame.  The longer duration between checks of a set the more packs are added that will not be checked.  The set size is growing.

I think subset=n/t feature is really designed to be used if you have multiple machines and using clustered storage, you could then run the check concurrently on _t_ number of machines.  It is not for a single machine to check.

For a single machine, using a percentage (and understanding a 25% check will not be complete in 4 checks) is superior, as you'll be checking random packs and have a higher chance of finding the needle in the haystack.
